code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -162,20 +226,19 @@
 			overlays += "drying_rack_drying"
 
 /obj/machinery/smartfridge/drying_rack/proc/dry()
-	for(var/obj/item/weapon/reagent_containers/food/snacks/S in contents)
-		if(S.dry) continue
-		if(S.dried_type == S.type)
-			S.dry = 1
-			item_quants[S.name]--
-			S.name = "dried [S.name]"
-			S.color = "#AAAAAA"
-			S.loc = loc
-		else
-			var/D = S.dried_type
-			new D(loc)
-			item_quants[S.name]--
-			qdel(S)
-		return
+	for(var/datum/data/stored_item/I in item_records)
+		for(var/obj/item/weapon/reagent_containers/food/snacks/S in I.instances)
+			if(S.dry) continue
+			if(S.dried_type == S.type)
+				S.dry = 1
+				S.name = "dried [S.name]"
+				S.color = "#AAAAAA"
+				I.get_product(loc)

```

Comment:
Removes the first (last) product from I, not necessarily this specific S instance.